tests/integrationv2/common.py

```diff
@@ -38,39 +35,28 @@ def pq_enabled():
 
 class AvailablePorts(object):
     """
-    This iterator will atomically return the next number.
+    This iterator will atomically return the next available port.
     This is useful when running multiple tests in parallel
     that all need unique port numbers.
     """
 
-    def __init__(self, low=8000, high=30000):
-        worker_count = int(os.getenv('PYTEST_XDIST_WORKER_COUNT'))
-        chunk_size = int((high - low) / worker_count)
-
-        # If xdist is being used, parse the workerid from the envvar. This can
-        # be used to allocate unique ports to each worker.
-        worker = os.getenv('PYTEST_XDIST_WORKER')
-        worker_id = 0
-        if worker is not None:
-            worker_id = re.findall(r"gw(\d+)", worker)
-            if len(worker_id) != 0:
-                worker_id = int(worker_id[0])
-
-        # This is a naive way to allocate ports, but it allows us to cut
-        # the run time in half without workers colliding.
-        worker_offset = (worker_id * chunk_size)
-        base_range = range(low + worker_offset, high)
-        wrap_range = range(low, low + worker_offset)
-        self.ports = iter(itertools.chain(base_range, wrap_range))
-
+    def __init__(self):
         self.lock = threading.Lock()
 
+    def _get_available_port(self):
+        next_port = 0
+        with socket.socket() as s:
+            s.bind(('', next_port))
+            next_port = s.getsockname()[1]
```

Comment:
Have you considered doing this in s2nc/s2nd instead? Are you doing it here to avoid the framework losing track of which port is being used?

Reply:
Correct- this seemed more straightforward given the framework.